third_party/chromium/media/filters/chunk_demuxer.cc

```diff
@@ -1110,6 +1123,20 @@ base::TimeDelta ChunkDemuxer::GetWriteHead(const std::string& id) const {
   return iter->second[0]->GetWriteHead();
 }
 
+void ChunkDemuxer::SetSourceBufferStreamMemoryLimit(const std::string& id,
+                                                    size_t limit) {
+  DCHECK(source_state_map_.find(id) != source_state_map_.end());
```

Comment:
Do we need a lock here and in the function below?

Reply:
done.